### DISCUSSION TITLE
How to pass vars to raw?

### DISCUSSION THREAD

**Discussion by dejurin** [amannn/next-intl#174] [Q&A]
```JSON
{
	"common": {
		"data": {
			"title": "Hello {name}",
			"description": "Hi, this is {SEO}",
		}
	}
}
```

Something like this?

```js
t.raw('common.data', { name: 'world', SEO: 'example' })
```

**Comment by amannn** [comment]
Hey @dejurin and thanks for the question!

`raw` doesn't accept any arguments, as the intention of the function is to avoid any processing and just resolve the message as-is.

Can you expand a bit more on your example? I guess you have something more sophisticated in mind, since the particular example you're mentioning would work with just `t`.

**Reply by dejurin** [reply]
Yes, I would like to call an JSON object and fill it with template objects. 
As in the example above. To reduce the lines of code.

**Comment by dejurin** [comment]
I would also like the pass of data models as a value for interpolation

**interpolation**
```JSON
{
	"common": {
		"data": {
			"title": "Hello {name.1}",
			"description": "Hi, this is {SEO.description}",
		}
	}
}
```

```JS
const data =  { name: ['world', 'worlds'], SEO: {description: 'summary text' }}

t('common.data.title', data)
// result: Hello worlds

t('common.data.description', data)
// result: Hi, this is summary text
```
and
```JS
const data =  { name: ['world', 'worlds'], SEO: {description: 'summary text' }}

t('common.data', data)
// result object: {title: 'Hello worlds', description: 'summary text'}

```
**Nasted**

Also the ability to refer to a repeated meaning, so as not to write the same words in different places.

More: https://github.com/formatjs/formatjs/issues/1270

**Reply by amannn** [reply]
`next-intl` currently supports the same feature set as the Format.JS parser, so this is currently not planned.

**Reply by dejurin** [reply]
OK, thanks for answer.
